packages/flutter/lib/src/rendering/shifted_box.dart

```diff
@@ -617,8 +633,10 @@ class RenderConstrainedOverflowBox extends RenderAligningShiftedBox {
     );
   }
 
+  // If shrinkWrap, the size will be as small as its child when non-overflowing,
+  // thus is cannot be sizedByParent.
   @override
-  bool get sizedByParent => true;
+  bool get sizedByParent => !shrinkWrap;
 
   @override
   Size computeDryLayout(BoxConstraints constraints) {
```

Comment:
What about this implementation?

Reply:
I am not very sure about the question. I add tests testing the case now